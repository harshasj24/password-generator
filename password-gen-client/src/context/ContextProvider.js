import React, {
  createContext,
  useContext,
  useEffect,
  useMemo,
  useState,
} from "react";

export const passwordCotext = createContext();

export const ContextProvider = ({ children }) => {
  const [sliderVal, setSliderVal] = useState(12);
  const [password, setPassword] = useState("");
  const [breakPoint, setBreakPoint] = useState(null);
  const [isLoading, setisloading] = useState(false);
  const breakPoints = {
    mobile: "(max-width:600px)",
    tablet: "(min-width:600px) and (max-width:960px)",
    laptop: "(min-width:960px) and (max-width:1023px)",
    desktop: "(min-width:1024px)",
  };

  const copyToClipBoard = (text) => {
    window.navigator.clipboard.writeText(text);
  };

  const matchMediaQuerry = (breakPoints, setBreakPoint) => {
    Object.keys(breakPoints).map((key) => {
      if (window.matchMedia(breakPoints[key]).matches) {
        setBreakPoint(key);
      }
    });
  };

  const breakPointObserver = () => {
    window.addEventListener("resize", () => {
      matchMediaQuerry(breakPoints, setBreakPoint);
    });
  };

  const values = useMemo(() => {
    return {
      password,
      setPassword,
      sliderVal,
      setSliderVal,
      copyToClipBoard,
      breakPointObserver,
      breakPoint,
      isLoading,
      setisloading,
    };
  });

  return (
    <div>
      <passwordCotext.Provider value={values}>
        {children}
      </passwordCotext.Provider>
    </div>
  );
};

export const usePassword = () => {
  return useContext(passwordCotext);
};
