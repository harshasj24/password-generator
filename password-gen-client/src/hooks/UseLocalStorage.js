import React, { useEffect, useState } from "react";
import CryptoJS from "crypto-js";

const encrypt = (data) => {
  const cyphertext = CryptoJS.AES.encrypt(JSON.stringify(data)).toString();
  return cyphertext;
};
const decrypt = (cyphertext) => {
  const bytes = CryptoJS.AES.decrypt(cyphertext);
  const originalData = JSON.parse(bytes.toString(CryptoJS.enc.Utf8));
  return originalData;
};
export const useLocalStorage = (key, defaultValue) => {
  const [value, setValue] = useState(() => {
    try {
      const localStorageVal = localStorage.getItem(key);
      if (localStorageVal) {
        return JSON.parse(localStorageVal);
      } else {
        localStorage.setItem(key, JSON.stringify(defaultValue));
        return defaultValue;
      }
    } catch (error) {
      return defaultValue;
    }
  });
  const setNewValue = (newValue) => {
    setValue(newValue);
    try {
      localStorage.setItem(key, JSON.stringify(newValue));
    } catch (error) {}
  };
  return [value, setNewValue];
};
