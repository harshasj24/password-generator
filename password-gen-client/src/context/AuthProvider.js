import React, {
  createContext,
  useContext,
  useEffect,
  useMemo,
  useState,
} from "react";
import { useNavigate } from "react-router-dom";
import { useLocalStorage } from "../hooks/UseLocalStorage";
import axios from "axios";
import { useVault } from "./VaultProvider";
const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
  // const navigate = useNavigate();
  // const [isLoggedIn, setIsLoggedin] = useState(false);
  const [user, setUser] = useLocalStorage("user", null);
  useEffect(() => {
    console.log(user);
  }, [user]);
  const login = async (userDetails) => {
    setUser(userDetails);
    console.log(userDetails, "auth-provider");
  };
  const logout = async () => {
    setUser(null);
    // console.log(responce);
  };
  const value = useMemo(() => {
    return { user, login, logout, setUser };
  }, [user, login]);
  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};

export const useAuth = () => {
  return useContext(AuthContext);
};
