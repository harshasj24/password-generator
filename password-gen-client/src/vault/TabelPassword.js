import {
  Button,
  CircularProgress,
  IconButton,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from "@mui/material";
import React, { useEffect, useState } from "react";
import LaunchIcon from "@mui/icons-material/Launch";

import { formatDate, formatTime } from "./customdate";
import { usePassword } from "../context/ContextProvider";
import { Link, useNavigate } from "react-router-dom";
import Launch from "@mui/icons-material/Launch";
import { useVault } from "../context/VaultProvider";
import axios from "axios";
import { useAuth } from "../context/AuthProvider";
import EditIcon from "@mui/icons-material/Edit";
import DeleteIcon from "@mui/icons-material/Delete";
import EditOutlinedIcon from "@mui/icons-material/EditOutlined";
import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
import CircularProgressWithLabel from "./CircularProgressWithLabel";
import { Facebook, NavigateBefore, Twitter } from "@mui/icons-material";
import { links, checkLinks } from "./constant";
import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
const TabelPassword = ({ passwords, setModal, search, handleModalOpen }) => {
  const { copyToClipBoard } = usePassword();
  const { updatePass, isUpdating, headers, setSelectedId, getOnePassword } =
    useVault();

  const navigate = useNavigate();
  const [currentDate, setCurrentDate] = useState(new Date());
  const getPercentage = (value) => {
    return (value / 50) * 100;
  };

  const handelClick = (_id) => async (e) => {
    // console.log(_id);
    // e.target.innerText = "....";
    // const responce = await axios.get(
    //   `http://localhost:4500/vault/getPassword/${_id}`,
    //   {
    //     headers: headers,
    //   }
    // );
    // copyToClipBoard(responce.data.data.password);
    // e.target.innerText = "Copied";
    // setTimeout(() => {
    //   e.target.innerText = "copy";
    // }, 2000);
    // // <CircularProgress size="23px" />
  };
  useEffect(() => {
    let interval = setInterval(() => {
      setCurrentDate(new Date());
    }, 60000);
    return () => {
      clearInterval(interval);
    };
  }, []);
  const modifieDate = (date) => {
    const newDate = new Date(date);

    var diff = Math.trunc(
      (currentDate.getTime() - newDate.getTime()) / 1000 / 60
    );

    if (diff < 1) {
      return "Just now";
    } else if (diff > 0 && diff <= 60) {
      return diff + "min ago";
    } else if (diff > 60 && diff < 1440) {
      return Math.trunc(diff / 60) + "hr ago";
    } else if (diff >= 1440) {
      return `${formatDate(newDate)}\n${formatTime(newDate)}`;
    }
  };

  const handleLinkClick = (_id, pName) => () => {
    isUpdating({ _id: _id, action: "edit", pName: pName });
    navigate("/access");
  };
  const getColor = (status) => {
    console.log(status);
    return status >= 12
      ? "success"
      : status > 5 && status <= 10
      ? "warning"
      : "error";
  };

  const handleDelete = (_id) => () => {
    isUpdating({ _id: _id, action: "delete" });
    navigate("/access");
  };
  return (
    <div>
      <TableContainer sx={{ height: "410px" }} component={Paper}>
        <Table stickyHeader>
          <TableHead sx={{ color: "white" }}>
            <TableRow>
              <TableCell>
                {" "}
                <h6>Website</h6>{" "}
              </TableCell>
              <TableCell>
                <h6> password</h6>{" "}
              </TableCell>
              <TableCell>
                {" "}
                <h6>Last Modified</h6>{" "}
              </TableCell>
              <TableCell>
                {" "}
                <h6>Status</h6>{" "}
              </TableCell>
              <TableCell>
                <h6>Actions</h6>{" "}
              </TableCell>
            </TableRow>
          </TableHead>
          <TableBody data-testid="tableBody">
            {passwords.filter((data) =>
              data.pName.toLowerCase().includes(search.toLowerCase())
            ).length > 0 ? (
              passwords
                .filter((data) =>
                  data.pName.toLowerCase().includes(search.toLowerCase())
                )
                .map((data, i) => (
                  <TableRow key={i}>
                    <TableCell>
                      {data.pName}
                      {checkLinks(data.pName) && (
                        <a
                          className="ml-2"
                          target="_blank"
                          href={checkLinks(data.pName)?.link}
                        >
                          <LaunchIcon fontSize="inherit" />
                        </a>
                      )}
                    </TableCell>
                    <TableCell>
                      <Button
                        onClick={handleModalOpen(data._id)}
                        data-testid={`copy${i}`}
                        size="small"
                        variant="outlined"
                      >
                        view/copy
                      </Button>
                      {/* <IconButton
                      onClick={handleModalOpen(data._id)}
                      data-testid={`copy${i}`}
                      size="small"
                      variant="outlined"
                    >
                      <MoreHorizIcon />
                    </IconButton> */}
                    </TableCell>
                    <TableCell>{modifieDate(data.date)}</TableCell>
                    <TableCell>
                      <CircularProgressWithLabel
                        color={getColor(data.status)}
                        value={getPercentage(data.status)}
                      />
                    </TableCell>
                    <TableCell>
                      <IconButton
                        size="small"
                        onClick={handleLinkClick(data._id, data.pName)}
                      >
                        <EditOutlinedIcon color="primary" fontSize="small" />
                      </IconButton>
                      <IconButton
                        onClick={handleDelete(data._id)}
                        sx={{ ml: 2 }}
                        size="small"
                      >
                        <DeleteOutlineIcon color="error" fontSize="small" />
                      </IconButton>
                      {/* <Button onClick={handleLinkClick(data._id)}>
                    change <Launch fontSize="inherit" />
                  </Button> */}
                    </TableCell>
                  </TableRow>
                ))
            ) : (
              <TableRow>
                <TableCell colSpan={5} className="text-center">
                  No Data Found
                </TableCell>
              </TableRow>
            )}
          </TableBody>
        </Table>
      </TableContainer>
    </div>
  );
};

export default TabelPassword;
