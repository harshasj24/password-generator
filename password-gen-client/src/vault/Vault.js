import React, { useEffect, useRef, useState } from "react";
import CustomModel from "./CustomModel";
import { useVault } from "../context/VaultProvider";
import TabelPassword from "./TabelPassword";
import vaultImage from "../assets/Image1.jpeg";
import "./vaultstyle.css";
import { usePassword } from "../context/ContextProvider";
import GetPassModel from "./GetPassModel";
import SearchIcon from "@mui/icons-material/Search";
import {
  Modal,
  Button,
  Card,
  CardHeader,
  IconButton,
  TextField,
  SwipeableDrawer,
  InputAdornment,
  FormControl,
  InputLabel,
  Snackbar,
  Alert,
} from "@mui/material";
import { ErrorMessage, Form, Formik } from "formik";
import * as yup from "yup";
import SwipeableEdgeDrawer from "./SwipeableEdgeDrawer";
import VaultCard from "./VaultCard";
import LockIcon from "@mui/icons-material/Lock";
import { useAuth } from "../context/AuthProvider";
import { LockOpen, LockOutlined } from "@mui/icons-material";
import DoneIcon from "@mui/icons-material/Done";
import CloseIcon from "@mui/icons-material/Close";
import Visibility from "@mui/icons-material/Visibility";
import VisibilityOff from "@mui/icons-material/VisibilityOff";
import AddPasswords from "./AddPasswords";
import { useNavigate } from "react-router-dom";

const Vault = () => {
  const {
    passwords,
    updatePass,
    getAllPasswords,
    checkPassword,
    getOnePassword,
    snackbar,
    handleSnackbar,
    modalDetails,
    handleModalDetails,
    savePassword,
  } = useVault();
  const { password, copyToClipBoard } = usePassword();
  const navigate = useNavigate();
  const isLoading = useRef(true);
  const { user } = useAuth();
  const [open, setOpen] = useState(false);

  const [search, setSearch] = useState("");
  const [confirmAccess, setConfirmAccess] = useState(false);
  const [grantedAccess, setGrantedAccess] = useState(false);
  const [loginPassword, setLoginPassword] = useState("");
  // const [modalDetails, setModalDetails] = useState({
  //   open: false,
  //   _id: null,
  //   type: "",
  // });
  const [vaultPassword, setVaultPassword] = useState({
    open: null,
    password: "",
  });

  const [typePassword, setTypePassword] = useState(true);
  const initialValues = {
    password: "",
    pName: "",
  };

  const validationSchema = yup.object({
    pName: yup.string().required("*Please provide a name for your password"),
    password: yup.string().required("password is required"),
  });
  const handelChange = (field, setFieldValues) => (e) => {
    setFieldValues(field, e.target.value);
    console.log(e.target.value);
  };
  const handleSubmit = (values) => {
    savePassword(values);
    handleClose();
    handleSnackbar(true, "Password Added Sucessfully");
  };

  const handleViewCopy = (type) => async () => {
    if (grantedAccess && !vaultPassword.password) {
      const responce = await getOnePassword(modalDetails._id);
      console.log(responce);
      if (responce.status === 200) {
        if (type === "view") {
          setVaultPassword({
            ...vaultPassword,
            open: true,
            password: responce.data.data.password,
          });
        } else if (type === "copy") {
          copyToClipBoard(responce.data.data.password);
          setVaultPassword({
            ...vaultPassword,
            password: responce.data.data.password,
          });
        }
      }
    } else {
      if (grantedAccess && type === "copy") {
        copyToClipBoard(vaultPassword.password);
      } else if (grantedAccess && type === "view") {
        setVaultPassword({
          ...vaultPassword,
          open: true,
        });
      } else {
        setConfirmAccess(true);
      }
    }
  };

  useEffect(() => {
    if (isLoading.current) {
      isLoading.current = false;
      getAllPasswords();
    }
  }, [snackbar]);
  const handleOpen = () => {
    setOpen(true);
  };
  const handleModalOpen = (_id) => () => {
    handleModalDetails({ ...modalDetails, open: true, _id, type: "view" });
  };
  const handleClose = () => {
    handleModalDetails({
      ...modalDetails,
      open: false,
      type: "",
    });
    setVaultPassword({ open: false, password: "" });
    setGrantedAccess(false);
    setConfirmAccess(false);
  };

  const handleDone = async () => {
    const payload = {
      email: user?.email,
      password: loginPassword,
    };
    const responce = await checkPassword(payload);
    if (responce) {
      setConfirmAccess(false);
      setGrantedAccess(true);
    } else {
      setConfirmAccess(true);
      alert("sede");
    }
  };

  const handleChange = (type) => (e) => {
    if (type === "search") {
      setSearch(e.target.value);
    }
    if (type === "login password") {
      setLoginPassword(e.target.value);
    }
  };

  const getLenth = (data, status) => {
    if (status === "strong") {
      console.log(data.filter((val) => val.status >= 12));
      return data.filter((val) => val.status >= 12).length;
    } else if (status === "week") {
      return data.filter((val) => val.status > 5 && val.status <= 10).length;
    } else if (status === "bad") {
      return data.filter((val) => val.status < 6).length;
    } else {
      return data.length;
    }
  };
  const passCard = [
    {
      title: "All",
      length: getLenth(passwords, ""),
      color: "#9fccf5",
    },
    {
      title: "Strong",
      length: getLenth(passwords, "strong"),
      color: "#73ba72",
    },
    {
      title: "Week",
      length: getLenth(passwords, "week"),
      color: "#f67c54",
    },
    {
      title: "Bad",
      length: getLenth(passwords, "bad"),
      color: "#f56551",
    },
  ];
  const action = (
    <React.Fragment>
      <IconButton size="small" aria-label="close" color="inherit">
        <CloseIcon fontSize="small" />
      </IconButton>
    </React.Fragment>
  );
  return (
    <div className="vault--wrapper">
      <div className="container">
        <div className="vault-header">
          <h3>Welcome, {user?.fName + " " + user.lName}</h3>
          <div className="search-filed">
            <TextField
              size="small"
              label=" "
              placeholder="search..."
              InputProps={{
                startAdornment: (
                  <InputAdornment position="start">
                    <SearchIcon />
                  </InputAdornment>
                ),
              }}
              InputLabelProps={{
                shrink: false,
              }}
              onChange={handleChange("search")}
            />
          </div>
        </div>
        <div className="vault-cards-section p-2">
          {passCard.map((val) => {
            return (
              <VaultCard
                title={val.title}
                length={val.length}
                color={val?.color}
              />
            );
          })}
        </div>

        <div className="vault-body__table mt-2">
          <TabelPassword
            search={search}
            passwords={passwords}
            setModal={setOpen}
            handleModalOpen={handleModalOpen}
          />
        </div>
      </div>
      {/* <div className="banner">
        <h2 className="text-light mt-3">Welcome To Vault</h2>
        <Button
          className="but"
          variant="outlined"
          sx={{ width: "180px" }}
          onClick={handleOpen}
        >
          Show Passwords
        </Button>
      </div>
      <div className="table">
        {/* <TabelPassword passwords={passwords} setModal={setOpen} /> */}
      {/* <SwipeableEdgeDrawer
          open={open}
          handleClose={handleClose}
          handleOpen={handleOpen}
        />
      </div>  */}

      <Modal open={modalDetails.open}>
        <div className="vault-modal--wrapper">
          {modalDetails.type === "view" && (
            <div className="vault-modal">
              <div className="vault-modal__header">Password Options</div>
              <div className="vault-modal__body">
                <div className="body__icons d-flex">
                  {!confirmAccess ? (
                    <div>
                      {grantedAccess ? (
                        <LockOpen color="success" sx={{ fontSize: "3rem" }} />
                      ) : (
                        <LockOutlined color="error" sx={{ fontSize: "3rem" }} />
                      )}
                    </div>
                  ) : (
                    <div className="container-layout__column">
                      <p className="text-center">
                        Please confirm your password
                      </p>
                      <div className="text-field container-layout__row ">
                        <TextField
                          variant="outlined"
                          size="small"
                          label=" "
                          placeholder="password"
                          type={"password"}
                          onChange={handleChange("login password")}
                          InputLabelProps={{ shrink: false }}
                          InputProps={{
                            endAdornment: (
                              <InputAdornment>
                                <div>
                                  <IconButton onClick={handleDone}>
                                    <DoneIcon />
                                  </IconButton>
                                  <IconButton
                                    onClick={() => {
                                      setConfirmAccess(false);
                                    }}
                                  >
                                    <CloseIcon />
                                  </IconButton>
                                </div>
                              </InputAdornment>
                            ),
                          }}
                        />
                        {/* <IconButton onClick={handleDone}>
                        <DoneIcon />
                      </IconButton>
                      <IconButton
                        onClick={() => {
                          setConfirmAccess(false);
                        }}
                      >
                        <CloseIcon />
                      </IconButton> */}
                      </div>
                    </div>
                  )}
                </div>
                <div className="body__actions d-flex flex-column ">
                  {vaultPassword.open ? (
                    <div>
                      <TextField
                        type={typePassword ? "password" : "text"}
                        size="small"
                        disabled
                        InputProps={{
                          endAdornment: (
                            <InputAdornment>
                              <IconButton
                                onClick={() => {
                                  setTypePassword(!typePassword);
                                }}
                              >
                                {typePassword ? (
                                  <Visibility />
                                ) : (
                                  <VisibilityOff />
                                )}
                              </IconButton>
                            </InputAdornment>
                          ),
                        }}
                        InputLabelProps={{ shrink: false }}
                        label=" "
                      />
                    </div>
                  ) : (
                    <Button
                      onClick={handleViewCopy("view")}
                      size="small"
                      variant="contained"
                    >
                      View Password
                    </Button>
                  )}
                  <Button
                    onClick={handleViewCopy("copy")}
                    size="small"
                    variant="contained"
                    color="success"
                  >
                    Copy Password
                  </Button>
                </div>
              </div>
              <div className="vault-modal__footer">
                <Button
                  onClick={handleClose}
                  className="footer_btn"
                  size="small"
                  variant="contained"
                >
                  Close
                </Button>
              </div>
            </div>
          )}
          {modalDetails.type === "add" && (
            <div className="add__modal bg-light">
              <div className="add-modal__header">
                <p>Add Your Custom Password</p>
                <IconButton onClick={handleClose}>
                  <CloseIcon />
                </IconButton>
              </div>
              <div className="add-modal__body">
                <Formik
                  initialValues={initialValues}
                  validationSchema={validationSchema}
                  onSubmit={handleSubmit}
                  enableReinitialize
                >
                  {({ setFieldValue }) => {
                    return (
                      <Form>
                        <div className="form__body">
                          <TextField
                            className=" my-3"
                            variant="outlined"
                            label="password"
                            name="password"
                            onChange={handelChange("password", setFieldValue)}
                            fullWidth
                          />
                          <ErrorMessage name="password">
                            {(msg) => {
                              return <p className="text-danger">{msg}</p>;
                            }}
                          </ErrorMessage>

                          <div className="pass-names my-3">
                            <TextField
                              onChange={handelChange("pName", setFieldValue)}
                              variant="outlined"
                              label="Name"
                              name="pName"
                              className="w-100"
                              fullWidth
                            />
                            <ErrorMessage name="pName">
                              {(msg) => {
                                return <p className="text-danger">{msg}</p>;
                              }}
                            </ErrorMessage>
                          </div>
                        </div>
                        <div className="vault-modal__footer">
                          <Button
                            type="submit"
                            className="d-flex ml-auto"
                            color="secondary"
                            variant="contained"
                          >
                            add
                          </Button>
                        </div>
                      </Form>
                    );
                  }}
                </Formik>
              </div>
            </div>
          )}
        </div>
      </Modal>

      <Snackbar
        autoHideDuration={6000}
        open={snackbar.open}
        onClose={() => {
          handleSnackbar(false, "");
        }}
        anchorOrigin={{
          horizontal: "center",
          vertical: "top",
        }}
      >
        <Alert action={action} severity="success">
          {snackbar.message}
        </Alert>
      </Snackbar>
    </div>
  );
};

export default Vault;
