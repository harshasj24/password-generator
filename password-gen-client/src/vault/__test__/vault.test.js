import { cleanup, render, renderHook, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { AuthProvider, useAuth } from "../../context/AuthProvider";
import { ContextProvider } from "../../context/ContextProvider";
import { VaultProvider } from "../../context/VaultProvider";
import TabelPassword from "../TabelPassword";
import Vault from "../Vault";
import axios from "axios";
import { useLocalStorage } from "../../hooks/UseLocalStorage";

let component = null;
let mock;

const passwords = [
  {
    pName: "G pay",
    _id: "1234ytfhsd567",
    date: new Date(),
  },
  {
    pName: "paytm",
    _id: "1234ytfhsd56785t",
    date: new Date(),
  },
  {
    pName: "phone pay",
    _id: "1234ytfhsd569w",
    date: new Date(),
  },
];

beforeEach(() => {
  component = (
    <BrowserRouter>
      <AuthProvider>
        <VaultProvider>
          <ContextProvider>
            <Vault />
          </ContextProvider>
        </VaultProvider>
      </AuthProvider>
    </BrowserRouter>
  );
});
afterEach(() => {
  cleanup();
});
describe("testing vault component", () => {
  it("it should render the component", () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("it should call get all paswords API", () => {
    const { user, setUser } = renderHook(() =>
      useLocalStorage("user", { id: 1223 })
    );
    const wrapper = render(component);
    axios.get.mockResolvedValueOnce({
      data: {
        data: passwords,
      },
    });
    expect(axios.get).toHaveBeenCalledTimes(1);
  });
});
