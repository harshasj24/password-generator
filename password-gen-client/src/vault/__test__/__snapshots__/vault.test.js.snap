// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing vault component it should render the component 1`] = `
<div>
  <div
    className="vault--wrapper"
  >
    <div
      className="container"
    >
      <div
        className="vault-header"
      >
        <h3>
          Welcome, 
          undefined undefined
        </h3>
        <div
          className="search-filed"
        >
          <div
            className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r0:"
              id=":r0:-label"
            >
               
            </label>
            <div
              className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <div
                className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
              >
                <span
                  className="notranslate"
                >
                  â€‹
                </span>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </div>
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1ua80n0-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r0:"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="search..."
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-1ftyaf0"
                >
                  <span>
                     
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="vault-cards-section p-2"
      >
        <div
          className="card--wrapper"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root d-card css-1ajek8z-MuiPaper-root-MuiCard-root"
          >
            <div
              className="card-content"
            >
              <div
                className="left-content"
              >
                <h6>
                  All
                </h6>
                <p>
                  Password
                </p>
              </div>
              <div
                className="right-content"
              >
                0
              </div>
            </div>
          </div>
        </div>
        <div
          className="card--wrapper"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root d-card css-1mngge0-MuiPaper-root-MuiCard-root"
          >
            <div
              className="card-content"
            >
              <div
                className="left-content"
              >
                <h6>
                  Strong
                </h6>
                <p>
                  Password
                </p>
              </div>
              <div
                className="right-content"
              >
                0
              </div>
            </div>
          </div>
        </div>
        <div
          className="card--wrapper"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root d-card css-kq901-MuiPaper-root-MuiCard-root"
          >
            <div
              className="card-content"
            >
              <div
                className="left-content"
              >
                <h6>
                  Week
                </h6>
                <p>
                  Password
                </p>
              </div>
              <div
                className="right-content"
              >
                0
              </div>
            </div>
          </div>
        </div>
        <div
          className="card--wrapper"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root d-card css-f5enr4-MuiPaper-root-MuiCard-root"
          >
            <div
              className="card-content"
            >
              <div
                className="left-content"
              >
                <h6>
                  Bad
                </h6>
                <p>
                  Password
                </p>
              </div>
              <div
                className="right-content"
              >
                0
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="vault-body__table mt-2"
      >
        <div>
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-v0tj71-MuiPaper-root-MuiTableContainer-root"
          >
            <table
              className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
              role={null}
            >
              <thead
                className="MuiTableHead-root css-197n1ah-MuiTableHead-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                     
                    <h6>
                      Website
                    </h6>
                     
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    <h6>
                       password
                    </h6>
                     
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                     
                    <h6>
                      Last Modified
                    </h6>
                     
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                     
                    <h6>
                      Status
                    </h6>
                     
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    <h6>
                      Actions
                    </h6>
                     
                  </th>
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                data-testid="tableBody"
                role={null}
              >
                <tr
                  className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium text-center css-1ex1afd-MuiTableCell-root"
                    colSpan={5}
                  >
                    No Data Found
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
